R/utils-get_code_dependency.R

```diff
@@ -163,17 +172,24 @@ extract_occurrence <- function(calls_pd) {
     calls_pd,
     function(call_pd) {
       # Handle data(object)/data("object")/data(object, envir = ) independently.
-      data_call <- call_pd$token == "SYMBOL_FUNCTION_CALL" & call_pd$text == "data"
+      data_call <- is_symbol(call_pd, "data")
       if (any(data_call)) {
```

Comment:
I think it's safe to do 
```suggestion
      if (any(is_symbol(call_pd, "data"))) {
```

By the way, it is vital that `is_symbol` always return a logical vector of length > 0, never `NULL`, `logical(0)`, `NA`. Please make sure it always happens in the context the function is used or introduce some argument checks to lock it in.

Reply:
`data_call` is used further after the `if` statement so I will keep it as a variable, and not a check in the `if`.

```r
      data_call <- is_symbol(call_pd, "data")
      if (any(data_call)) {
        sym <- call_pd[which(data_call) + 1, "text"]
```

Reply:
> By the way, it is vital that is_symbol always return a logical vector of length > 0, never NULL, logical(0), NA. Please make sure it always happens in the context the function is used or introduce some argument checks to lock it in.

I don't think there is no need to check that. `is_symbol` returns vector of logicals of length 0 or greater. Then it is used in `any` that returns `FALSE` on `logical(0)` which is great, and `TRUE` on `logical(n)` if any element is equal to `TRUE`. I don't think we need any special treatment of edge cases as it's vectorized logical operation on two vectors

`call_pd$token == "SYMBOL_FUNCTION_CALL" & call_pd$text == symbol`

and those vectors can either be empty or of type `character` (and never seen it to be `NA`).

This was already tested in the battlefield in `tests/` folder anyway : )

Just to show what are the results of usage please see below, but I think testing this is an overkill. I didn't put anything extra in the function besides `checkmate::check_string(symbol)`

```r
call_pd_1 <- data.frame(
 token = character(0),
 text = character(0)
)

call_pd_2 <- data.frame(
 token = 'SYMBOL_FUNCTION_CALL',
 text = 'data1'
)

call_pd_3 <- data.frame(
 token = c('SYMBOL', 'SYMBOL_FUNCTION_CALL'),
 text = c('iris', 'data')
)

is_symbol(call_pd_1, 'data')
> logical(0)
is_symbol(call_pd_2, 'data')
> [1] FALSE
is_symbol(call_pd_3, 'data')
> [1] FALSE  TRUE
any(is_symbol(call_pd_1, 'data'))
> [1] FALSE
any(is_symbol(call_pd_2, 'data'))
> [1] FALSE
any(is_symbol(call_pd_3, 'data'))
> [1] TRUE
any(NULL)
> [1] FALSE
any(logical(0))
> [1] FALSE

```

Reply:
Ok, I ended up with such checks for `is_symbol` functions

```r
checkmate::check_string(symbol)
checkmate::check_data_frame(call_pd)
checkmate::check_names(call_pd, must.include = c('token', 'text'))
 ```
 
 I think it's enough for this simple function.

Reply:
Fair enough, 
```
> any(NULL)
[1] FALSE
> any(logical(0))
[1] FALSE
```

Reply:
> `data_call` is used further

You're right, I missed that.